web/wp-content/plugins/tmbi-prebid/js/tmbi-prebid.js

```diff
@@ -0,0 +1,319 @@
+					}
+				},
+				bids: []
+			};
+			prebid_ad = wp.hooks.applyFilters( 'ad_slot_to_prebid_ad', prebid_ad, ad_slot, options );
+			return prebid_ad;
+		}
+
+		self.get_bids = function( slots ) {
+			return new Promise( function( resolve, reject, notify ) {
+				var pbto = setTimeout( function(){
+					console.log( 'Ad Stack', 'Prebid get_bids timeout', prebid_timeout );
+					resolve();
+				}, prebid_timeout );
+
+
+				pbjs.que = pbjs.que || [];
+				pbjs.que.push(function(){
+					console.log( 'Ad Stack', 'Prebid addAdUnits' );
+					pbjs.addAdUnits( slots.map( ad_slot_to_prebid_ad ) );
+					console.log( 'Ad Stack', 'Prebid requestBids' );
+					pbjs.requestBids({
+						bidsBackHandler: function( bids ) {
+							clearTimeout( pbto );
+							console.log( 'Ad Stack', 'Got Prebid bids' );
+							got_bids = true;
+							resolve( {
+								bidder: 'Prebid.js',
+								bids: bids
+							} );
+						},
+						timeout: prebid_timeout
+					});
+				});
+			} );
+		}
+
+		self.set_bids = function() {
+			console.log( 'Ad stack', 'Trying to set bids for Prebid' );
+			return new Promise( function( resolve, reject, notify ) {
+				if ( got_bids ) {
+					pbjs.que.push( function(){
+						console.log( 'Ad Stack', 'Prebid bids set' );
+						pbjs.setTargetingForGPTAsync();
+						resolve();
+					} );
+				} else {
+					console.log( 'Ad Stack', 'Prebid bids not set' );
+					resolve();
+				}
+			} );
+		}
+	};
+
+	bidders.push( new Prebid( pbjs ) );
+
+	return bidders;
+} );
+
+/* @todo: separate this into different files for dev, then combine again for prod */
+wp.hooks.addFilter( 'ad_slot_to_prebid_ad', function add_rubicon_bid_params( prebid_ad, ad_slot, options ) {
+	if ( typeof prebid_conf !== 'undefined' && prebid_conf['rubicon_blocked_flag'] ) {
+		console.warn('Rubicon blocked from url');
+		return prebid_ad;
+	}
+	if ( ! ( options.rubicon_site_id && options.rubicon_account_id && options.rubicon_atf_zone_id && options.rubicon_btf_zone_id ) ) {
+		console.warn( 'Ad Stack', 'Prebid', 'Rubicon is not configured.' );
+		return prebid_ad;
+	}
+	console.log( 'Ad Stack', 'Prebid', 'Adding Rubicon bid data' );
+	prebid_ad.bids.push({
+		bidder: 'rubicon',
+		params: {
+			accountId: options.rubicon_account_id,
+			siteId: options.rubicon_site_id,
+			zoneId: ( ad_slot.targeting.tf[0] == 'atf' ? options.rubicon_atf_zone_id : rubicon.rubicon_btf_zone_id )
+		}
+	});
+	return prebid_ad;
+} );
+
+wp.hooks.addFilter( 'ad_slot_to_prebid_ad', function add_openx_bid_params( prebid_ad, ad_slot, options ) {
+	if ( typeof prebid_conf !== 'undefined' && prebid_conf['openx_blocked_flag'] ) {
+		console.warn('OpenX blocked from url');
+		return prebid_ad;
+	}
+	if ( ! ( options.openx_del_domain ) ) {
+		console.warn( 'Ad Stack', 'Prebid', 'OpenX is not configured / is not enabled.' );
+		return prebid_ad;
+	}
+	console.log( 'Ad Stack', 'Prebid', 'OpenX call for', ad_slot );
+	/* @todo: Find a sane way to make this matrix configurable */
```

Comment:
This PR seems a good time to remove the TOH, RD and FHM code, and to create a ticket to fix this. Is there one?

Reply:
@fariasf  Yes i think of moving this to settings page for prebid.
Till now didnt create. will have to create one.

Reply:
@fariasf https://readersdigest.atlassian.net/browse/DRDT-104 created